### ISSUE TITLE
[Other] Transmission

### ISSUE THREAD

**Issue by Anuskuss** [PBH-BTN/PeerBanHelper#683]
### 问题描述 - Issue Description

I understand that after #382 Transmission is considered deprecated and when compared to something like v6.0.2 the latest version doesn't do anything. But is it really that much of a support burden? If I set up my Transmission to restart once a day (so it can get the latest blocklist) that would work perfectly fine for me. I know that means that some blocked peers will continue downloading (unless you manually merge https://github.com/transmission/transmission/pull/7167) but I think that's better than nothing. And while the advanced anti-leeching features are great (hopefully https://github.com/transmission/transmission/pull/7172 will help in that regard) I don't really need them. So can you please restore the ability to populate the blocklist and just let the user know that they have to restart Transmission manually?

### 检查清单 - Check list

- [X] 这不是一个错误 （BUG） (This is not an bug/error)
- [X] PeerBanHelper 已更新到最新版本，非最新版本不接受任何错误反馈，任何非最新版本的 Issue 将被 立 刻 关 闭，不会有人给您提供任何支持 (I'm running the latest version of PBH that can be found in Github Relases, non-latest release won't receive any support)
- [X] 我已检查过 [PBH 文档](https://docs.pbh-btn.com/)（特别是常见问题），且即使使用了搜索也没有找到与此有关的内容 (This not a question/or the question that not listed in README's FAQ or [PBH WIKI](https://docs.pbh-btn.com/))
- [X] 我同意遵守 PBH-BTN 包容性条约，不发布 “嘲讽、骂战、引战、开盒（有时也称为人肉搜索）、人身攻击、仇恨、暴力、侮辱性言辞、违法违规、黑灰产、危害国家安全、实施或帮助他人实施电信犯罪” 等内容。并已知晓如果仍旧发布了这些内容，我的账号将立刻从包括但不限于 PBH-BTN 组织、社交软件中封禁。所有主题、内容都将被立刻删除或折叠，撤销、删除和收回您所做出的一切贡献，并封禁 BTN 网络的中账号权限、排除您所提交的所有数据。在您违反相关规则时，PBH-BTN 将会将您的注册、登录、和最近访问的 IP 地址、电子邮件地址、以及其它可能追踪您或将您去匿名化的信息从定期删除转为永不删除，并在任何国家或地区的政府、公安机关或有关部门需要时无通知的提供这些数据。 (I agree to abide by the PBH-BTN Inclusivity Pact by not posting content such as “taunting, name-calling, war-mongering, open-boxing (sometimes referred to as mansplaining), personal attacks, hatred, violence, insulting language, illegal activities, black and grey business, endangering national security, and committing or assisting others in committing telecommunication crimes”. I am aware that if I continue to post such content, my account will be immediately banned from organizations including but not limited to PBH-BTN, social software. All topics and content will be immediately deleted or collapsed, all contributions will be revoked, deleted and retracted, and you will be banned from the BTN network and all data you have submitted will be excluded. In the event of a violation of these rules, PBH-BTN will delete your registration, login, and most recent IP address, email address, and any other information that may be used to track you or de-anonymize you from regular to permanent deletion, and will make this data available to the government, public security, or other relevant authorities without notice if they request it, no matter what country or region.)

**Comment by Ghost-chu** [comment]
The code for the Transmission feature is still in the PeerBanHelper codebase, but it has been disabled and not updated for more than six months due to the issues we mentioned, and we don't expect to maintain support for Transmission in the future.
However, you can indeed re-enable it manually before it is completely removed.

You can follow the documentation to manually re-enable and manually add the configuration (it is disabled in the WebUI so edit the file is necessary).

https://pbh-btn.github.io/pbh-docs/docs/downloader/Transmission#%E9%87%8D%E5%90%AF%E7%94%A8%E8%A2%AB%E7%A6%81%E7%94%A8%E7%9A%84-transmission

I'd love to restore active support for Transmission, but it's going to be hard to do that until Transmission makes changes.

**Comment by Anuskuss** [comment]
Okay thanks for the link, that indeed made it work. And it also populated `blocklist-url` although I'm wondering how often that will happen?

So the warning says
> Warning: Use Transmission Adapter is discourage. Frequent starting and stopping of torrents on seeds that are often subject to bans can result in frequent updates to the tracker server, indirectly triggering DoS attacks. This increases the load on the tracker server and may lead to your IP address being banned by the tracker. We encourage you to migrate to other downloaders whenever possible.

How can I disable this "Frequent starting and stopping" (if it's not already disabled) so I can just restart Transmission once a day myself? (I also found a bug when `ignore-private` is set to `true` but since you don't officially support Transmission anymore I'm not going to ask you for a fix.)

**Comment by Ghost-chu** [comment]
> Okay thanks for the link, that indeed made it work. And it also populated `blocklist-url` although I'm wondering how often that will happen?
> 
> So the warning says
> 
> > Warning: Use Transmission Adapter is discourage. Frequent starting and stopping of torrents on seeds that are often subject to bans can result in frequent updates to the tracker server, indirectly triggering DoS attacks. This increases the load on the tracker server and may lead to your IP address being banned by the tracker. We encourage you to migrate to other downloaders whenever possible.
> 
> How can I disable this "Frequent starting and stopping" (if it's not already disabled) so I can just restart Transmission once a day myself? (I also found a bug when `ignore-private` is set to `true` but since you don't officially support Transmission anymore I'm not going to ask you for a fix.)

This is actually one of the problems we're having, PeerBanHelper is designed to disconnect immediately after a block, and if it doesn't disconnect, it tries again and again. Each repeated ban is also logged as a banning operation thus polluting the statistics.
I would suggest that you could use [BTN-Collected-Rules](https://raw.githubusercontent.com/PBH-BTN/BTN-Collected-Rules/refs/heads/main/combine/all.txt) as we are constantly updating it with data from BTN and it works just fine. Transmission 3.0+ should read this rule set without any problems.

Of course the best option is always to switch to another supported downloader.

**Comment by Anuskuss** [comment]
Also don't forget about the ability to just straight up block peers through `iptables` which means that you wouldn't have to use Transmission's blocklist at all (the rules apply immediately). Is this already trackable without running into the "repeated ban is also logged as a banning operation" situation?

**Comment by Ghost-chu** [comment]
> > I would suggest that you could use [BTN-Collected-Rules](https://raw.githubusercontent.com/PBH-BTN/BTN-Collected-Rules/refs/heads/main/combine/all.txt)
> 
> This seems like a fair compromise but then you're really forced to restart Transmission yourself. Would you consider undeprecating Transmission and removing the "stop-and-start" workaround once [transmission/transmission#7167](https://github.com/transmission/transmission/pull/7167) is merged? Or maybe add a `restart_torrent` option in the config that defaults to `true` which a user then can manually disable? I already build my own Transmission with a few unmerged patches so it would be easy to apply 7167 myself. I know that Transmission has been stuck in limbo for a few months but I really think we can make this work.
> 
> > Of course the best option is always to switch to another supported downloader.
> 
> I tested the big 3 (Transmission, qBittorrent, Deluge) many many years ago and for me the clear winner was Transmission for servers (low CPU/memory usage, better headless) and qBittorrent for desktops (more features, better speeds). Back then `qbittorrent-nox` was quite bad and you didn't have have a good remote GUI but maybe that has improved since then? Anyway I have hundreds of torrents in Transmission and I really don't wanna switch unless I have to. I'll consider qBittorrent once Transmission officially dies but until then I hope we can make this work. The dev of 7167 said that `blocklist-update` will trigger a blocklist refresh and 7167 will let you kick blocked peers so I think it's possible to make Transmission a first-class citizen again.
> 
> P.S. Is there any way to change the loglevel? My systemd service/journal is flooded with Java nonsense and I would really like to only see warnings and errors there (the rest can go into `/etc/peerbanhelper/logs/latest.log`). I'm doing `java -jar PeerBanHelper.jar | grep --line-buffered -v '\(/\||-\)INFO\|^ '` but there's got to be a better way.

Hi I want to know which one deploy method that you have to use to setup PBH? The .sh script or .deb package or something else?

**Comment by Ghost-chu** [comment]
> Also don't forget about the ability to just straight up block peers through `iptables` which means that you wouldn't have to use Transmission's blocklist at all (the rules apply immediately). Is this already trackable without running into the "repeated ban is also logged as a banning operation" situation?

This works for Linux systems, but there doesn't seem to be a better way for Windows.
I've tried operating the firewall but it doesn't work well, especially since sometimes the number of IPs blocked reaches the upper limit of a single rule.

Another important reason is that I don't think I should have PeerBanHelper running under root or Adninistrator privilege levels, whereas operating the firewall obviously requires elevated privileges.

**Comment by Anuskuss** [comment]
> Hi I want to know which one deploy method that you have to use to setup PBH? The .sh script or .deb package or something else?

The .deb package.

> This works for Linux systems, but there doesn't seem to be a better way for Windows.

I don't know about Windows but that's why I was asking for a `restart_torrent` option. So the user can decide for themself if they want PBH to restart the torrent, or either restart the client or use a firewall instead.

> Another important reason is that I don't think I should have PeerBanHelper running under root

In Linux land it already runs as user `peerbanhelper`. For use with `iptables` you'd also need at least `CAP_NET_RAW` and `CAP_NET_ADMIN` (haven't tested).

**Comment by Ghost-chu** [comment]
> The .deb package.

@Gaojianli Can you redirect stdout/stderr to a file in /var/log for .deb packaging?

> I don't know about Windows but that's why I was asking for a restart_torrent option. So the user can decide for themself if they want PBH to restart the torrent, or either restart the client or use a firewall instead.

If Transmission can merge that PR, we can actually not restart Torrent but just update the blocklist url.

**Comment by Ghost-chu** [comment]
For what it's worth, our community members have created a tool that can proxy the Transmission API and the System Firewall API and convert them to the qBittorrent API and provide it to PeerBanHelper or any other program that needs the qBittorrent API (although it's not yet complete) if you're interested. If you are interested, you can check it out.

Disclaimer: This project is not affiliated with PeerBanHelper, PBH-BTN or me.

https://github.com/azicen/transmission-proxy

**Comment by Anuskuss** [comment]
> Can you redirect stdout/stderr to a file in /var/log for .deb packaging?

The full log is already being recorded in `/etc/peerbanhelper/logs/latest.log` so the only thing that would make sense it to set `StandardOutput=null`. But I need a filtered log in systemd so I can quickly check for problems (e.g. `systemctl status peerbanhelper`).

> If Transmission can merge that PR, we can actually not restart Torrent but just update the blocklist url.

Transmission could merge that PR but it's release cycle is currently stuck. So the merged PR could sit in `master` for months and nobody would benefit. A `restart_torrent` option would fix it *today* but I guess the code would be dead once Transmission 4.0.7/4.1 releases so I understand why you're maybe opposed to that. It's your call after all.

**Comment by Anuskuss** [comment]
> our community members have created a tool that can proxy the Transmission API and the System Firewall API and convert them to the qBittorrent API

That sounds really interesting but I hope we can make it work with the tools we have available now.

**Comment by Anuskuss** [comment]
I saw that you disabled the logs completely now but I still wanna see them when I do `systemctl status peerbanhelper` so instead I just turned on the loglevel (#698). There are still a bunch of useless lines like
```
22:25:01,641 |-INFO in ch.qos.logback.classic.LoggerContext[default] - This is logback-classic version 1.5.12
```
but it's better than before.

**Comment by Ghost-chu** [comment]
> The loglevels seem a bit arbitrary, no? Because why aren't these `INFO`s?

This warning should not be triggered under normal circumstances, as indicated when the PBH reports the problem:

1. BanList update failed
2. Downloader disconnect Peer lagging
3. user manually changed BanList

Reapplying the full list is a fix, but it sometimes doesn't always work, especially since some qBittorrent occasionally stops processing the BanList while it is running, and the user must be instructed that when a large number of WARNs and ERRORs recur, there must be some kind of problem.

**Comment by Anuskuss** [comment]
Ah okay, I misunderstood then: The error happens because the peer is already blocked but the client is still connecting to it. I guess I'll merge https://github.com/transmission/transmission/pull/7167 myself and see if it still happens.

**Comment by azicen** [comment]
> For what it's worth, our community members have created a tool that can proxy the Transmission API and the System Firewall API and convert them to the qBittorrent API and provide it to PeerBanHelper or any other program that needs the qBittorrent API (although it's not yet complete) if you're interested.You can check it out at: https://github.com/azicen/transmission-proxy无论如何，我们的社区成员创建了一个工具，可以代理传输 API 和系统防火墙 API，并将它们转换为 qBittorrent API，并将其提供给 PeerBanHelper 或任何其他需要 qBittorrent API 的程序（尽管尚未完成） ）如果您感兴趣。您可以查看：https: [//github.com/azicen/transmission-proxy](https://github.com/azicen/transmission-proxy)
> 
> Disclaimer: This project is not affiliated with PeerBanHelper, PBH-BTN or me.免责声明：该项目与 PeerBanHelper、PBH-BTN 或我无关。

在我尝试构建这个工具来正常使用 Transmission 时遇到了问题，即使通过防火墙中断连接，Transmission 也不会立刻更新数据快照。这会引起 PBH 的不满，尝试通过更多的 api 校验客户端数据并更新所有 banlist，这回增大 Transmission 客户端的负担，导致 Transmission 性能下降。

结论就是，即使使用防火墙拦截，也无法达到一个可用的预期，我认为还是需要 Transmission 解决阻断问题。


When I attempted to build this tool for normal use of Transmission, I encountered issues where Transmission does not immediately update the data snapshot even when the connection is interrupted through the firewall. This can cause dissatisfaction with PBH, as it tries to validate client data through more API checks and update all banlists, which increases the load on the Transmission client, leading to a decline in Transmission's performance.

The conclusion is that even with the use of a firewall to intercept, it is not possible to achieve a usable expectation. I believe that it is still necessary to use Transmission to BanList issue.

**Comment by Anuskuss** [comment]
Thanks for trying. I'm not sure if I wanna give master a go. The whole v4 line has not been great, it's always 1 step forwards 2 steps back. I wish they would just release a new minor version so many people can test it. Buidling master myself (and spoofing the version to the whitelisted v4.0.5) unfortunately bears the risk of getting me banned (from private trackers).

**Comment by Anuskuss** [comment]
> so I'm resorting to `StandardOutput=null` and `RuntimeMaxSec=1d` for now.

I initially did this because I would quite often run into these pause/unpause loops because Transmission would (somehow) miss the blocklist update which results in a 100s of torrents (even my private ones because the `ignore-private` is currently broken) getting paused every 2 minutes (my update interval is 1 minute). I did try `transmission-remote --blocklist-update` but that didn't work and I'm not sure if https://github.com/transmission/transmission/pull/7167 would even help in this case because to me it feels like Transmission just fails to load new rules (after some time?) altogether.

Anyway I just realized that Transmission has the ability to [reload](https://github.com/transmission/transmission/blob/002c8912167142ea3794d4e927fbb0c9d4b3edce/docs/Release-Notes.md?plain=1#L1016) via `SIGHUP` so instead of killing my daemon once a day I'll just reload it every hour or so. Doing it like that results in 0 downtime.

**Comment by Anuskuss** [comment]
Transmission seems to be broken in v7.2. I don't expect you to fix it or even care so I'll be staying on v7.1.5 (until Transmission finally releases a new version).

<details><summary>Log</summary>

```
[00:33:28] [main/WARN]: PeerBanHelper SQLite Connection Pool - keepaliveTime is greater than or equal to maxLifetime, disabling it.
[00:33:29] [main/WARN]: [Restricted] Due to Transmission's RPC-API limitations, the PeerId blacklist function and the excessive download module of ProgressCheatBlocker are unavailable
[00:33:29] [main/WARN]: Warning: Use Transmission Adapter is discourage. Frequent starting and stopping of torrents on seeds that are often subject to bans can result in frequent updates to the tracker server, indirectly triggering DoS attacks. This increases the load on the tracker server and may lead to your IP address being banned by the tracker. We encourage you to migrate to other downloaders whenever possible. https://github.com/PBH-BTN/PeerBanHelper/issues/382
[00:33:33] [virtual-507/ERROR]: Failed to execute module Progress Cheat Blocker, please report to PeerBanHelper developer
java.lang.NullPointerException: Cannot invoke "java.lang.Long.longValue()" because the return value of "cordelia.rpc.types.Torrents.getSizeWhenDone()" is null
	at com.ghostchu.peerbanhelper.downloader.impl.transmission.TRTorrent.getCompletedSize(TRTorrent.java:45)
	at com.ghostchu.peerbanhelper.module.impl.rule.ProgressCheatBlocker.shouldBanPeer(ProgressCheatBlocker.java:232)
	at com.ghostchu.peerbanhelper.PeerBanHelperServer.checkBan(PeerBanHelperServer.java:847)
	at com.ghostchu.peerbanhelper.PeerBanHelperServer.lambda$checkBans$23(PeerBanHelperServer.java:639)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.lang.VirtualThread.run(Unknown Source)
[00:33:33] [virtual-914/ERROR]: An exception occurred while ban Peer, specific Peer unable to ban, Please report these information to PeerBanHelper developer
java.lang.NullPointerException: Cannot invoke "java.lang.Long.longValue()" because the return value of "cordelia.rpc.types.Torrents.getSizeWhenDone()" is null
	at com.ghostchu.peerbanhelper.downloader.impl.transmission.TRTorrent.getCompletedSize(TRTorrent.java:45)
	at com.ghostchu.peerbanhelper.wrapper.TorrentWrapper.<init>(TorrentWrapper.java:24)
	at com.ghostchu.peerbanhelper.wrapper.PeerMetadata.<init>(PeerMetadata.java:26)
	at com.ghostchu.peerbanhelper.wrapper.BanMetadata.<init>(BanMetadata.java:25)
	at com.ghostchu.peerbanhelper.PeerBanHelperServer.lambda$banWave$10(PeerBanHelperServer.java:570)
	at java.base/java.util.concurrent.Executors$RunnableAdapter.call(Unknown Source)
	at java.base/java.util.concurrent.FutureTask.run(Unknown Source)
	at java.base/java.lang.VirtualThread.run(Unknown Source)
```
</details>

**Comment by Ghost-chu** [comment]
> Transmission seems to be broken in v7.2. I don't expect you to fix it or even care so I'll be staying on v7.1.5 (until Transmission finally releases a new version).
> 
> Log

Transmission code still exists in the repository, because we still want to restore support for Transmission.  
But since Transmission has not merged https://github.com/transmission/transmission/pull/7167, it is actually very unusable. Therefore, none of our test volunteers test commits related to Transmission.

I just completed a fix for this issue, but I haven't tested it yet.

**Comment by Anuskuss** [comment]
Thanks, 3e1803a3b8b79a15ae9147ad6ed7a870cbcb6f47 fixed it indeed.

> But since Transmission has not merged

Yeah I'm also thinking about switching clients because Transmission seems to be close to death but who knows, maybe it can make a comeback.

**Comment by Ghost-chu** [comment]
I saw Transmission merged https://github.com/transmission/transmission/pull/7167 into their 4.1.0-beta2 version.

I'm re-evaluating Transmission availability.

**Comment by Anuskuss** [comment]
I think once https://github.com/transmission/transmission/pull/7172 and https://github.com/transmission/transmission/pull/7514 get merged this issue can be closed.

**Comment by Ghost-chu** [comment]
I saw both of them get merged, So I implemented them, but we still need to wait for Transmission to release a new test version with these patches before we can test them.


https://github.com/PBH-BTN/PeerBanHelper/commit/969257961702f62bc0e77d31ae9a07db29abba20  
https://github.com/PBH-BTN/PeerBanHelper/commit/ec927e2de8364a40d0cf2f0bbff74af8e043b9bd

**Comment by Anuskuss** [comment]
Both PRs are in v4.1.0b4 now (together with https://github.com/transmission/transmission/pull/6936 and https://github.com/transmission/transmission/pull/7173). Will this already work with the latest version of PBH or do you need to add a `Transmission >= 4.1.0b4` check first?

Also if you do do a new release, could you please change
https://github.com/PBH-BTN/PeerBanHelper/blob/becff75c1e80806daf8db18a58cc755450b66af4/pkg/deb/DEBIAN/control#L4
to
```suggestion
Depends: java-runtime (>=25) | java-runtime-headless (>=25), libc6
```

Thanks.

**Comment by Ghost-chu** [comment]
> Both PRs are in v4.1.0b4 now (together with [transmission/transmission#6936](https://github.com/transmission/transmission/pull/6936) and [transmission/transmission#7173](https://github.com/transmission/transmission/pull/7173)). Will this already work with the latest version of PBH or do you need to add a `Transmission >= 4.1.0b4` check first?

I just tested Transmission v4.1.0-beta4, but this version appears to have some issues preventing me from testing RPC operations.   

I've just opened an issue in Transmission's repository: https://github.com/transmission/transmission/issues/7885 . Let's keep an eye on it.

> 
> Also if you do do a new release, could you please change
> 
> [PeerBanHelper/pkg/deb/DEBIAN/control](https://github.com/PBH-BTN/PeerBanHelper/blob/becff75c1e80806daf8db18a58cc755450b66af4/pkg/deb/DEBIAN/control#L4)
> 
> Line 4 in [becff75](/PBH-BTN/PeerBanHelper/commit/becff75c1e80806daf8db18a58cc755450b66af4)
> 
>  Depends: java-runtime (>=25), libc6 
> 
> to
> ```
> Depends: java-runtime (>=25) | java-runtime-headless (>=25), libc6
> ```
> 
> Thanks.

Done. ;)

**Comment by Anuskuss** [comment]
Since https://github.com/transmission/transmission/issues/7885 seems to be a non-issue I decided to upgrade both Transmission and PeerBanHelper to see if it all works now. I'm running both on my server and qBittorrent on my desktop and PeerId-based blocking seems to work but it's unfortunately broken because it has banned my *own* desktop:
> Script PeerID/ClientName 伪装检查: Peer reporting: PeerId=lxfcnte####################=, ClientName=qbittorrent/5.1.4, But PBH excepted=-qb

I think `peer_id` is base64-encoded and PBH expects a *string* (that starts with `-qb`) so right now it will ban **every client**.

No idea if statistics-based (`bytes_to_client` and `bytes_to_peer`) is working.

**Comment by Anuskuss** [comment]
I removed `tr_base64_encode()` in [libtransmission/rpcimpl.cc#L662](https://github.com/transmission/transmission/blob/fcd1e2ff8a1535f69f6d3f9e365c6a952e7d1206/libtransmission/rpcimpl.cc#L662) for now until the value gets handled properly. Anyway here are my findings:
1. *Uploaded* and *Downloaded* per client statistics show up, so I'll presume that statistics-based blocking will work (haven't seen it in the log yet).
2. Everytime I restart Transmission (and PBH, they start together) I see
> Failed to update IP blocklist for Transmission
PeerBanHelper was unable to update the IP blocklist for Transmission because Transmission notified us that the blocklist update failed. Please check and ensure that the "Settings -> Basic Settings -> WebUI -> Prefix" configuration item is correctly set to point to the PeerBanHelper WebUI address and that it is accessible and reachable by Transmission (merely confirming access via a browser is insufficient). If this configuration is incorrect, Transmission will not be able to properly block or unblock any IP addresses.
>
> If you need any assistance with the configuration, please refer to the support documentation or request help in the community.

It seems to be working though.

3. It looks like you don't issue a `blocklist_update` after unbanning a client. I unbanned my own desktop client and didn't see neither a `blocklist_url` or `blocklist_size` change. Manually running `transmission-remote --blocklist-update` decreased `blocklist_size` by 1.
4. My Transmission client couldn't find my qBittorrent client after unbanning + `blocklist_update`. I don't know if it was just a coincidence that it worked after a restart but that's ought to be a Transmission issue though.

**Comment by Ghost-chu** [comment]
I haven't had a chance to conduct any tests yet; it's been an extremely busy period.
I shall look into these changes as soon as I have the opportunity.

**Comment by Ghost-chu** [comment]
I took some time to continue testing Transmission and added the appropriate code to restore Transmission support. Unfortunately, I encountered a new issue: https://github.com/transmission/transmission/issues/8074 .This should continue to prevent Transmission support from being available in the next version.

**Comment by Anuskuss** [comment]
> added the appropriate code to restore Transmission support

Does that mean I have to wait for v9.2.4? Because I've been running Transmission v4.1.0b + PBH v9.2.3 for a few weeks now and it seems to be working fine (I think).

> I encountered a new issue: https://github.com/transmission/transmission/issues/8074

I hit the same issue and I don't think it's a big deal. Let's hope it gets fixed before v4.1.0 drops but in the meantime I'd suggest you add a Transmission-specific workaround where you send at least 1 entry (e.g. `0.0.0.0-0.0.0.0`) when the list is empty.

**Comment by Anuskuss** [comment]
v4.1.0 is out now. PBH still needs a fix for the new base64-encoded `peer_id` and as of now still needs to maintain the "empty blocklist"-workaround (see https://github.com/PBH-BTN/PeerBanHelper/issues/683#issuecomment-3658098975).
